service/http/dffml_service_http/routes.py

```diff
@@ -157,37 +186,82 @@ async def multicomm_dataflow(self, config, request):
-                        not input_data["definition"]
-                        in config.dataflow.definitions
-                    ):
-                        return web.json_response(
-                            {
-                                "error": f"Missing definition for {input_data['definition']} in dataflow"
-                            },
-                            status=HTTPStatus.NOT_FOUND,
+            # Accept a list of input data according to config.input_mode
+            if config.input_mode == "default":
+                # TODO validate that input data is dict of list of inputs each item
+                # has definition and value properties
+                for ctx, client_inputs in (await request.json()).items():
+                    for input_data in client_inputs:
+                        if (
+                            not input_data["definition"]
+                            in config.dataflow.definitions
+                        ):
+                            return web.json_response(
+                                {
+                                    "error": f"Missing definition for {input_data['definition']} in dataflow"
+                                },
+                                status=HTTPStatus.NOT_FOUND,
+                            )
+                    inputs.append(
+                        MemoryInputSet(
+                            MemoryInputSetConfig(
+                                ctx=StringInputSetContext(ctx),
+                                inputs=[
+                                    Input(
+                                        value=input_data["value"],
+                                        definition=config.dataflow.definitions[
+                                            input_data["definition"]
+                                        ],
+                                    )
+                                    for input_data in client_inputs
+                                ],
+                            )
                         )
+                    )
+            elif ":" in config.input_mode:
+                preprocess_mode, input_def = config.input_mode.split(":")
+                if input_def not in config.dataflow.definitions:
+                    return web.json_response(
+                        {
+                            "error": f"Missing definition for {input_data['definition']} in dataflow"
+                        },
+                        status=HTTPStatus.NOT_FOUND,
+                    )
+                if preprocess_mode == "json":
+                    value = await request.json()
+                elif preprocess_mode == "str":
+                    value = await request.text()
+                elif preprocess_mode == "bytes":
+                    value = await request.read()
+                # TODO : Verify usage of `stream`
+                elif preprocess == "stream":
+                    value = await request.content.read(-1)
```

Comment:
```suggestion
                    value = request.content
```

Reply:
We want pass the object that can be read (aka the [`StreamReader`](https://docs.aiohttp.org/en/stable/streams.html#aiohttp.StreamReader) as the value